docs/vector-library.md

```diff
@@ -0,0 +1,123 @@
+
+## Design
+
+The default metatable for vectors should now be the `vector` library. While buffers and coroutines do not have this, vectors likely should for ergonomics.
+
+### Library functions & constants
+
+This RFC proposes the following basic functions & constants as a starting point, but as with all builtin libraries, future RFCs can propose additional functions.
+
+---
+
+`vector(x: number, y: number, z: number)`
+
+Creates a vector with 3 components: x, y, z. If the feature flag for wide vectors is enabled, a fourth argument `w: number?` will be introduced.
+
+Due to the common usage of vectors, vector creation should be ergonomic. Therefore, it is probably worth breaking the `create()` naming standard.
+
+`vector.magnitude(vecA: vector): number`
+
+Calculates the magnitude of a given vector.
+
+`vector.normalized(vec: vector): vector`
+
+Returns the normalized version (aka unit vector) of a given vector. If a zero vector is passed, return zero. For further clarification, `vector.normalized(vector(0, 0, 0))` should return zero.
+
+`vector.cross(vecA: vector, vecB: vector): vector`
+
+Returns the cross product of two vectors. If 4-wide vectors are enabled, this function will ignore the fourth component, and return the 3-dimensional cross product.
+
+`vector.dot(vecA: vector, vecB: vector): number`
+
+Returns the dot product of two vectors.
+
+`vector.floor(vec: vector): vector`
+
+Equivalent of `math.floor` for vectors.
+
+`vector.ceil(vec: vector): vector`
+
+Equivalent of `math.ceil` for vectors.
+
+`vector.angle(vecA: vector, vecB: vector): number`
+
+Returns the angle between two vectors in radians.
+
+`vector.abs(vec: vector): vector`
+
+Applies math.abs component-wise for each vector. Equivalent of: `vector(math.abs(vec.x), ect)`.
+
+`vector.sign(vec: vector): vector`
+
+Applies `math.sign` component-wise for each vector. Equivalent of: `vector(math.sign(vec.x), ect)`.
+
+`vector.clamp(vec: vector, min: vector, max: vector): vector`
+
+Applies `math.clamp` component-wise for each vector. Equivalent of: `vector(math.clamp(vec.x, min.x, max.x), ect)`.
+
+`vector.max(...: vector): vector`
+
+Applies `math.max` component-wise for each vector. Equivalent of: `vector(math.max((...).x), ect)`.
+
+`vector.min(...: vector): vector`
+
+Applies `math.min` component-wise for each vector. Equivalent of: `vector(math.min((...).x), ect)`.
+
+---
+
+`vector.zero`
+
+Vector where `x=0, y=0, z=0, w?=0`.
+
+`vector.one`
+
+Vector where `x=1, y=1, z=1, w?=1`.
+
+---
+
+### Arithmetic operations
+
+Primitive operators for vectors are already implemented, so this RFC doesn't concern vector arithmetic.
+
+### Native codegen
+
+In the future, vectors will have special treatment in native codegen. This creates an expectation that the vector library will also have special treatment in codegen, but it isn't clear what that will look like, or if the performance benefits make sense.
```

Comment:
This can be removed as vectors already have special treatment in codegen.

Generally, library proposals shouldn't have to think about fastcalls/lowering hooks as those are implementation details which are implemented without a proposal being required.
Whether or not existing custom lowering hooks from Roblox will be included (or written from scratch) doesn't have to be specified.

Reply:
Sounds good